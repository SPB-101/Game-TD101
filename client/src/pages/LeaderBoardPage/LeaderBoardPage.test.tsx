import React from "react";
import renderer from "react-test-renderer";
import { StaticRouter } from "react-router";

import { LeaderBoardPage } from "./index";

jest.mock("./LeaderBoardItem", () => ({
  LeaderBoardItem() {
    return null;
  },
}));

jest.mock("../../component/Pagination", () => ({
  Pagination() {
    return `pagination component`;
  },
}));

describe("<LeaderBoardPage />", () => {
  it("should render correctly", () => {
    const tree = renderer
      .create(
        <StaticRouter location="/leaderboard">
          <LeaderBoardPage />
        </StaticRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
