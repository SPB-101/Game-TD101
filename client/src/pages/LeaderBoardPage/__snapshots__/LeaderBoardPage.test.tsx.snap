// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LeaderBoardPage /> should render correctly 1`] = `
<Fragment>
  <Link
    className="button button_back leader-board__button"
    to="/menu"
  >
    backToMenu
  </Link>
  <Wrapper
    className="leader-board"
    size="xl"
  >
    <h1
      className="leader-board__title"
    >
      leaderboard
    </h1>
    <ul
      className="leader-board__list"
    >
      <LeaderBoardItem
        key="0"
        playerImage="https://uifaces.co/our-content/donated/7JZz4G7I.jpg"
        playerName="Kristina Trifunov"
        rankingPosition={1}
        score={12123232}
      />
      <LeaderBoardItem
        key="1"
        playerImage="https://images.generated.photos/QbcqC7vbdYI9lI-q6inuEaCPd_yC4v7OkPMqCdWk-tM/rs:fit:512:512/Z3M6Ly9nZW5lcmF0/ZWQtcGhvdG9zLzA5/OTk2NjMuanBn.jpg"
        playerName="Vivian Painter"
        rankingPosition={2}
        score={12323}
      />
      <LeaderBoardItem
        key="2"
        playerImage="https://images.generated.photos/rF3Ir6Y3AZiiqNshk_D1rTEAUsgYz4wv_CLWd0EJZmM/rs:fit:512:512/Z3M6Ly9nZW5lcmF0/ZWQtcGhvdG9zL3Yy/XzA1NzM2MTUuanBn.jpg"
        playerName="Aniah Lassiter"
        rankingPosition={3}
        score={9898}
      />
      <LeaderBoardItem
        key="3"
        playerImage="https://images.generated.photos/KSkFZay8lawllgpfawH1UjaofmrXm1sPLZiauTWaYag/rs:fit:512:512/Z3M6Ly9nZW5lcmF0/ZWQtcGhvdG9zLzAw/MzYyNTIuanBn.jpg"
        playerName="Krish Kendall"
        rankingPosition={4}
        score={3490}
      />
      <LeaderBoardItem
        key="4"
        playerImage=""
        playerName="Dave Smiths"
        rankingPosition={5}
        score={1233}
      />
    </ul>
    <Pagination
      pageLimit={2}
      totalRecords={5}
    />
  </Wrapper>
</Fragment>
`;
